### DISCUSSION TITLE
proposal on http client

### DISCUSSION THREAD

**Discussion by Ousret** [eternnoir/pyTelegramBotAPI#2249] [Ideas]
Hello,

I saw that this project depend on requests for http interactions and also support aiohttp in order to serve optional async interfaces.

Let me introduce you to [Niquests](https://github.com/jawah/niquests), it is a drop in replacement for Requests with both synchronous and asynchronous interface. Along with native support for HTTP/2 and 3, amongst other interesting things like but not limited to DoH, DoQ, OCSP checks, happy eyeballs ...

It should help simplifying your code base, you could automatically generate the async part from your sync code for example.

Let me know if you show any interest in this,
Would be happy to help or answer any concerns you may have.

regards,

**Comment by coder2020official** [comment]
This is your library :)
Nevertheless, if we plan to unify the package for API requests - I already have some libs in my mind to use. However, I doubt that we will ever switch to other libraries unless something extraordinary happens

**Reply by Ousret** [reply]
> This is your library :)

Indeed.

> I already have some libs in my mind to use. 

Plural?

> unless something extraordinary happens

Well. Too bad, if Niquests isn't appealing, idk what can.

Nevertheless, thanks for the response.
regards,
